runners/samza/src/main/java/org/apache/beam/runners/samza/translation/SamzaPipelineTranslator.java

```diff
@@ -107,6 +114,47 @@ public static void createConfig(
     pipeline.traverseTopologically(visitor);
   }
 
+  /**
+   * Builds a map from PTransform to its input and output PValues. The map is serialized and stored
+   * in the job config.
+   */
+  public static Map<String, Map.Entry<String, String>> buildTransformIOMap(
```

Comment:
Moved it to JSON renderer, it still needs to be a separate scan of the pipeline using SamzaPipelineVisitor instead of a generic Beam PipelineVisitor which is used by the JSONRenderer reason being SamzaPipelineVisitor traverses the pipeline differently (does not enter some composite transforms I if they can be directly translated) - PValues Maps i need to be populated dependent on this logic